metaseq/hub_utils.py

```diff
@@ -678,21 +678,29 @@ def generate(
                                 : max_tokens[i]
                             ]
 
-                    # cut off the starting token
+                    # cut off the starting </s> token and the first token, and attach
+                    # a None at the beginning.
+                    # So, if the tokens are </s>, The, nice, cat
+                    # scores are            -0.5, -0.3, -0.2, -0.1
+                    # We would remove the scores for </s> and the first word "The"
+                    # and attach a None to signify the score for "The"
+                    # This is in line with how the OpenAI API operates.
+                    assert tokens[0] == self.task.target_dictionary.eos_index
+
                     tokens_no_eos = tokens[1:] if echo else tokens
-                    scores_with_eos = [None] + scores[1:] if echo else scores
+                    scores_no_eos = [None] + scores[2:] if echo else scores
```

Comment:
eeee i'm not sure if this is the implementation we want

Reply:
I am with Stephen here. I prefer the model return scores[1:] here.